test/e2e/framework/gateways.go

```diff
@@ -183,17 +183,23 @@ func (f *Framework) DeleteGateway(cluster ClusterIndex, name string) {
 	}, NoopCheckResult)
 }
 
+func (f *Framework) SaveGatewayNodeForTest(cluster ClusterIndex, gwNode string) {
```

Comment:
This doesn't need to be public - it will be called internally by `DoFailover`. 

```suggestion
func (f *Framework) saveGatewayNode(cluster ClusterIndex, gwNode string) {
```

Reply:
Done